### PR TITLE
feat: Migrated module App/Jobs from CARML - `avm/res/app/job`

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

Migrated `Microsoft.App/jobs` from CARML & updated to latest specs

cc: @MrRoundRobin fyi

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.app.job](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.app.job.yml)      |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: .github/workflows/avm.res.app.job.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+name: "avm.res.app.job"
+
```
Comment (AlexanderSehr) [inline .github/workflows/avm.res.app.job.yml:2]: Blocking comment until MAR file is updated

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -7,10 +7,10 @@
 /avm/ptn/authorization/role-assignment/ @Azure/avm-ptn-authorization-roleassignment-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/ptn/security/security-center/ @Azure/avm-ptn-security-securitycenter-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/aad/domain-service/ @Azure/avm-res-aad-domainservice-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/aad/domain-service/ @Azure/avm-res-aad-domainservice-module-owners-bicep
 /avm/res/analysis-services/server/ @Azure/avm-res-analysisservices-server-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/api-management/service/ @Azure/avm-res-apimanagement-service-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/app/container-app/ @Azure/avm-res-app-containerapp-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/app/job/ @Azure/avm-res-app-job-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:13]: Blocking comment until adoption is confirmed

Comment (AlexanderSehr) [reply .github/CODEOWNERS:13]: That was quick. Thanks @matebarabas

File: .github/workflows/avm.res.app.job.yml (hunk 1 above)
Comment (matebarabas) [reply .github/workflows/avm.res.app.job.yml:2]: MAR manifest updated; you can resolve this now.

File: avm/res/app/job/main.bicep [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Container App.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Required. Resource ID of environment.')
+param environmentId string
```
Comment (eriqua) [inline avm/res/app/job/main.bicep:12]: @AlexanderSehr don't recall, did we eventually opt for having `resourceId` suffix instead of `Id`?

Comment (eriqua) [reply avm/res/app/job/main.bicep:12]: Apparently so https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr22---category-inputs---parametersvariables-for-resource-ids

Comment (AlexanderSehr) [reply avm/res/app/job/main.bicep:12]: We did, yes. Good call

Comment (AlexanderSehr) [reply avm/res/app/job/main.bicep]: Fixed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**